From 8b9d1117800de9a51c21725479eadb7ad0807220 Mon Sep 17 00:00:00 2001
From: Subbaraya Sundeep <sbhatta@marvell.com>
Date: Tue, 4 Aug 2020 11:57:53 +0530
Subject: [PATCH 0684/1921] octeontx2-pf: Avoid null pointer dereference

otx2_mbox_up_handler_cgx_ptp_rx_info can be
called for a VF too and VF do not have ptp
context. Hence check for valid ptp pointer
before dereferencing.

Change-Id: I30424bc1231ea4384868f22f896933fdab702c1d
Signed-off-by: Subbaraya Sundeep <sbhatta@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/33393
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[WK: The original patch got from Marvell sdk11.21.09]
Signed-off-by: Wenlin Kang <wenlin.kang@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
index 5fc5c01d8370..388254b6086c 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_pf.c
@@ -875,6 +875,9 @@ int otx2_mbox_up_handler_cgx_ptp_rx_info(struct otx2_nic *pf,
 {
 	int i;
 
+	if (!pf->ptp)
+		return 0;
+
 	pf->ptp->ptp_en = msg->ptp_en;
 
 	/* notify VFs about ptp event */
-- 
2.31.1

